runtime/runtime/src/lib.rs

```diff
@@ -1679,63 +1778,105 @@ impl Runtime {
         let apply_state = &mut processing_state.apply_state;
         let state_update = &mut processing_state.state_update;
 
-        let signed_txs =
-            processing_state.transactions.transactions.into_iter().cloned().collect::<Vec<_>>();
-        for (tx_hash, result) in Self::parallel_validate_transactions(
-            &apply_state.config,
-            apply_state.gas_price,
-            signed_txs,
-            apply_state.current_protocol_version,
-        ) {
-            match result {
-                Ok((validated_tx, cost)) => {
-                    let (receipt, outcome_with_id) = match self.process_transaction(
+        let tx_batches = TransactionBatches::new(processing_state.transactions.transactions);
+
+        let batch_outcomes = tx_batches
+            .par_batches()
+            .map(|batch| {
+                self.process_batched_transactions(
+                    batch,
+                    &apply_state,
+                    state_update,
+                    apply_state.gas_price,
+                    apply_state.block_height,
+                    apply_state.current_protocol_version,
+                    None,
+                )
+            })
+            .collect_vec_list();
+
+        let mut all_processed: Vec<(usize, ProcessedTransaction)> = Vec::new();
```

Comment:
```suggestion
        let mut all_processed = Vec::new();
```

I think the compiler can figure it out?